homeassistant/components/recorder/core.py

```diff
@@ -741,52 +750,55 @@ def _run(self) -> None:
             self.migration_is_live = migration.live_migration(schema_status)
 
         self.hass.add_job(self.async_connection_success)
-        database_was_ready = self.migration_is_live or schema_status.valid
-
-        if database_was_ready:
-            # If the migrate is live or the schema is valid, we need to
-            # wait for startup to complete. If its not live, we need to continue
-            # on.
-            self._activate_and_set_db_ready()
-
-            # We wait to start a live migration until startup has finished
-            # since it can be cpu intensive and we do not want it to compete
-            # with startup which is also cpu intensive
-            if self._wait_startup_or_shutdown() is SHUTDOWN_TASK:
-                # Shutdown happened before Home Assistant finished starting
+
+        # First do non-live migration steps, if needed
+        if not schema_status.valid:
```

Comment:
Existing problem: `.valid` is a confusing name here

Reply:
I agree, I was confused by it when implementing the changes in this PR. Do you have  a better suggestion? Maybe just have two bools `migration_needed` and `repair_needed`?

Reply:
> Maybe just have two bools `migration_needed` and `repair_needed`?

Yes, that is much more clear 👍

Reply:
As discussed on Discord, this should happen in a separate PR